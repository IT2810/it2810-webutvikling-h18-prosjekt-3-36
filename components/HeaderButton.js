import React from "react";

import { StyleSheet, Button } from 'react-native';
import Colors from "../constants/Colors";

export default class HeaderButton extends React.Component {
  
  render() {
    return <Button {...this.props} style={styles.button} background="transparent" color={Colors.tintColor}/>;
  }
}


const styles = StyleSheet.create({
  button: {
    fontWeight: "bold" 
  }
});