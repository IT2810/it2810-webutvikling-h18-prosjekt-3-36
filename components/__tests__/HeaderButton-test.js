import React from "react";
import ShallowRenderer from 'react-test-renderer/shallow';
import HeaderButton from "../HeaderButton";
import appState from '../../store/reducers/Reducer'
import { createStore } from 'redux';

const store = createStore(appState)

it("HeaderButton Matches snapshot.", () => {
    const renderer = new ShallowRenderer();
    renderer.render(<HeaderButton title="test" onPress={()=> {}} store={store} navigation={{ getParam: () => { }, setParams: () => { } }} isModalVisible={false} />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
});