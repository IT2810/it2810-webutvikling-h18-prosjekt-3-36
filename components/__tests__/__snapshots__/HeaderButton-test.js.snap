// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderButton Matches snapshot. 1`] = `
<Button
  background="transparent"
  color="#ff2d55"
  isModalVisible={false}
  navigation={
    Object {
      "getParam": [Function],
      "setParams": [Function],
    }
  }
  onPress={[Function]}
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
  style={
    Object {
      "fontWeight": "bold",
    }
  }
  title="test"
/>
`;
